File: 3200-maximum-height-of-a-triangle/3200-maximum-height-of-a-triangle.cpp
class Solution {
public:
    int count_odd(int x) {
        int level = 0;
        int sum = 1, c = 1;
        while (sum <= x) {
            level++;
            c += 2;
            sum += c;
        }
        return level;
    }
    int count_even(int x) {
        int level = 0;
        int sum = 2, c = 2;
        while (sum <= x) {
            level++;
            c += 2;
            sum += c;
        }
        return level;
    }
    int maxHeightOfTriangle(int red, int blue) { 
        int olr = count_odd(red); //odd leve of red
        int elr = count_even(red);//eve lev of red
        
        int olb = count_odd(blue); //odd leve of blue
        int elb = count_even(blue);//eve lev of blue

        int h1=0,h2=0;
        if(olr>elb){
            h1 = 2*elb+1;
        }else if(olr <= elb){
            h1 = 2*olr;
        }
        if(olb>elr){
            h2 = 2*elr+1;
        }else if(olb<=elr){
            h2 = 2*olb;
        }
        return max(h1,h2);
    }
};

File: 3200-maximum-height-of-a-triangle/3200-maximum-height-of-a-triangle.java
class Solution {
    public int maxHeightOfTriangle(int red1, int blue1) {
        if(red1==1 && blue1 == 1){
            return 1;
        }
        if(red1==1 || blue1==1){
            return 2;
        }
        int max = 0;
        for(int i=0;i<=1;i++){
            int height = 0;
            int blue = blue1;
            int red = red1;
            int f = i;
            boolean can_grow = true;
            while(can_grow){
                if(f==0 && blue>0 && (blue>=height+1)){
                    height++;
                    blue-=height;
                    f=1;
                    System.out.println("blue");
                }
                else if(f==1 && red>0 && (red>=height+1)){
                    height++;
                    red-=height;
                    f=0;
                    System.out.println("red");
                
                }else{
                    can_grow=false;
                }

            }
            System.out.println(height);
            if(max<height){
                max = height;
            }
        }
        return max;
    }
}